src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -162,9 +164,14 @@ public void setArchivesEntryBookFilePath(Path archivesEntryBookFilePath) {
     }
 
     @Override
-    public Optional<String> getOfflineLink(String url) {
-        return storage.getOfflineLink(url)
-                .map(path -> path.toUri().toString());
+    public Optional<URL> getOfflineLink(URL url) {
+        return storage.getOfflineLink(url).flatMap(path -> {
+            try {
+                return Optional.of(path.toUri().toURL());
+            } catch (MalformedURLException e) {
+                return Optional.empty();
```

Comment:
Catching this means that in the unlikely event that local offline links become malformed, we default back to the original remote URL, but we should not expect this to ever happen.